### PR TITLE
gh-117739: Update definition for global interpreter lock for 3.13 

### PR DESCRIPTION
**Pull request by savannahostrowski**
* Issue: gh-117739

----
📚 Documentation preview 📚: https://cpython-previews--117740.org.readthedocs.build/

### CODE REVIEW

File: Doc/glossary.rst [hunk 1]
```diff
@@ -547,12 +547,10 @@ Glossary
...
-      Past efforts to create a "free-threaded" interpreter (one which locks
-      shared data at a much finer granularity) have not been successful
-      because performance suffered in the common single-processor case. It
-      is believed that overcoming this performance issue would make the
-      implementation much more complicated and therefore costlier to maintain.
-
+      As of Python 3.13, the GIL can be disabled using the ``--disable-gil``
+      build configuration or ``PYTHON_GIL=0`` environment variable. This feature
+      enables improved performance for multi-threaded applications or on
+      multi-core systems. For more details, see :pep:`703`.
```
Comment (ericvsmith) [inline Doc/glossary.rst]: Is it true that disabling the GIL will result in improved performance for single-threaded applications on multi-core systems? That's what the "or" implies to me.

File: Doc/glossary.rst [hunk 2]
```diff
@@ -547,12 +547,10 @@ Glossary
...
       when doing I/O.
 
-      Past efforts to create a "free-threaded" interpreter (one which locks
-      shared data at a much finer granularity) have not been successful
-      because performance suffered in the common single-processor case. It
-      is believed that overcoming this performance issue would make the
-      implementation much more complicated and therefore costlier to maintain.
-
+      As of Python 3.13, the GIL can be disabled using the ``--disable-gil``
+      build configuration or ``PYTHON_GIL=0`` environment variable. This feature
```
Comment (hugovk) [inline Doc/glossary.rst]: This should link to https://docs.python.org/dev/using/configure.html#cmdoption-disable-gil and https://docs.python.org/dev/using/cmdline.html#envvar-PYTHON_GIL:

```suggestion
      As of Python 3.13, the GIL can be disabled using the :option:`--disable-gil`
      build configuration or :envvar:`PYTHON_GIL=0 <PYTHON_GIL>` environment variable. This feature
```

I think the text isn't quite right:

First Python must be configured and built with `--disable-gil`, _and_ then Python needs to be run with `PYTHON_GIL=0` (or `-X gil 0`).

File: Doc/glossary.rst (hunk 1 above)
Comment (savannahostrowski) [reply Doc/glossary.rst]: Thanks - I tried to make this a bit less ambiguous.

**Review by hugovk — APPROVED** [review]
Thanks!

@colesbury Please could you have a quick check and merge if it's fine?

**Review by ericvsmith — APPROVED** [review]

**Review by colesbury — APPROVED** [review]
Great!

**Pull request merged**